packages/react-native-sdk/docusaurus/docs/reactnative/06-advanced/02-chat-with-video.mdx

```diff
@@ -3,6 +3,483 @@ title: Chat Integration
+      } catch (e) {
+        console.error('[Chat client]: Failed to disconnect', e);
+      }
+      setChatClient(null);
+    };
+
+    return () => {
+      disconnectRef.current = cleanUp();
+    };
+  }, [apiKey, userData, tokenProvider]);
+
+  return chatClient;
+};
+```
+
+:::note
+The `apiKey`, `userData` and the `tokenProvider` comes up from the `ChatWrapper` created above.
+:::
+
+## Adding Chat into Video
+
+### Passing the `ChatWrapper`
+
+If we extend our [Video Calling Tutorial](../../tutorials/video-calling/), our `ChatWrapper` component can we wrapped as a parent/child of the `StreamVideo` as:
+
+```tsx
+import React, { useEffect, useState } from 'react';
+import {
+  Call,
+  StreamCall,
+  StreamVideo,
+  StreamVideoClient,
+} from '@stream-io/video-react-native-sdk';
+import { VideoCallUI } from './components/VideoCallUI';
+import { ChatWrapper } from './src/components/ChatWrapper';
+
+export default function App() {
+  // ...
+
+  return (
+    // highlight-next-line
+    <ChatWrapper>
+      <StreamVideo client={client}>
+        <StreamCall call={call}>
+          <VideoCallUI />
+        </StreamCall>
+      </StreamVideo>
+      // highlight-next-line
+    </ChatWrapper>
+  );
+}
+```
+
+### Creating the Chat window/screen
+
+After the chat client is wrapped to the application, we can focus on creating the chat window/screen. We will take the help of the [`Channel`](https://getstream.io/chat/docs/sdk/reactnative/core-components/channel/), [`MessageList`](https://getstream.io/chat/docs/sdk/reactnative/ui-components/message-list/) and the [`MessageInput`](https://getstream.io/chat/docs/sdk/reactnative/ui-components/message-input/) component to do so.
+
+We will also create a channel with the same id as the call id to make it distinct.
+
+:::note
+While creating a channel, please make sure the channel type is `videocall`. This will add the appropriate permissions/capabilites to the channel for video call by default.
+:::
+
+An example of the same is shown below:
+
+```tsx
+import React, { useEffect, useState } from 'react';
+import { SafeAreaView } from 'react-native';
+import {
+  Channel,
+  MessageInput,
+  MessageList,
+  useChatContext,
+} from 'stream-chat-react-native';
+import { AuthenticationProgress } from '../../components/AuthenticatingProgress';
+import { Channel as ChannelType } from 'stream-chat';
+import { StreamChatGenerics } from '../../../types';
+
+export const ChatScreen = () => {
+  const [channel, setChannel] = useState<
+    ChannelType<StreamChatGenerics> | undefined
+  >(undefined);
+  const { client } = useChatContext();
+
+  const CHANNEL_TYPE = 'videocall';
+  const CHANNEL_ID = 'your-call-id'; // You can get the call id through sharing params while routing through navigation.
+
+  useEffect(() => {
+    const createChannel = async () => {
+      const newChannel = await client.channel(CHANNEL_TYPE, CHANNEL_ID);
+      setChannel(newChannel);
```

Comment:
cant we just watch here instead of `useChannelWatch`

Reply:
this simplifies it

Reply:
also why await here? 

and there is no need to create the channel ?

Reply:
and there is no need to create the channel ?

Reply:
client.channel creates a channel automatically if not created. We created the channel already in the ChatScreen

Reply:
https://getstream.io/chat/docs/javascript/creating_channels/?language=javascript#1.-creating-a-channel-using-a-channel-id

Reply:
@khushal87 you have linked, but please read the link

we must call `channel.create()`

Reply:
I don't think it's needed for channels with `videocall` permissions. Anyone can send message to it, that's its speciality

Reply:
Resolved since we call `channel.watch()`, we discussed yesterday.